Support downloading list of files in Resteasy Reactive Multipart

Sgitario opened the pull request:
Additionally, it adds support of DownloadFile objects. 

Fix https://github.com/quarkusio/quarkus/issues/26164

Sgitario commented:
/cc @eye0fra @geoand

geoand commented on independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/multipart/PartItem.java:
```diff
@@ -24,4 +26,8 @@ public MediaType getType() {
         return value;
     }
+
+    public String getRawGenericType() {
```
I don't think this name properly conveys what this does.
At the very least, we should probably document what it returns for both generic and non generic cases.

Sgitario replied:
Renamed to "firstParamType" and added comments in the method.

Sgitario replied:
Let me know if the name looks better for you

geoand reviewed (commented):
LGTM, just added one comment

Sgitario committed:
Support downloading list of files in Resteasy Reactive Multipart

Additionally, it adds support of DownloadFile objects. 

Fix https://github.com/quarkusio/quarkus/issues/26164

geoand reviewed (approved):
LGTM, thanks a lot!

eye0fra commented:
Awesome! Thanks a lot! @Sgitario do you have a sample of the raw response of the ### multi-file download?

Sgitario commented:
> Awesome! Thanks a lot! @Sgitario do you have a sample of the raw response of the ### multi-file download?

You can see the raw response by running this test: https://github.com/quarkusio/quarkus/pull/26309/files#diff-136f284323fc8e3022c3f73dce4ed1d4e1ce5d4adf1c6e9affe221f6b0c6befdR102

Either read the `response` String, or add `.log().all()` after the line 106.
